core/src/main/scala/kafka/log/LogSegment.scala

```diff
@@ -161,13 +161,13 @@ class LogSegment private[log] (val log: FileRecords,
       if (largestTimestamp > maxTimestampSoFar) {
         maxTimestampAndOffsetSoFar = TimestampOffset(largestTimestamp, shallowOffsetOfMaxTimestamp)
       }
+      bytesSinceLastIndexEntry += records.sizeInBytes
```

Comment:
Well, it seems that the physicalPosition we add to the index corresponds to the position before the records are appended. So, updating bytesSinceLastIndexEntry after the index append seems correct.